tools/cache_creator/units/cache_creator_tests.cpp

```diff
@@ -128,3 +128,108 @@ TEST_CASE("Full UUID roundtrip") {
   CHECK(cc::hexStringToUuid(hexStr, dumped));
   CHECK(dumped == uuid);
 }
+
+TEST_CASE("Get cache info from invalid PAL metadata blob") {
+  char badMetadata[10] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+
+  llvm::StringRef noteBlob(badMetadata, 10);
+  llvm::Expected<cc::ElfLlpcCacheInfo> cacheInfoOrErr = cc::getCacheInfoFromMetadataBlob(noteBlob);
+
+  CHECK(cacheInfoOrErr.errorIsA<llvm::StringError>());
+
+  llvm::Error err = cacheInfoOrErr.takeError();
+  llvm::consumeError(std::move(err));
+}
+
+TEST_CASE("Get cache info from valid PAL metadata blob") {
+  // Sample AMDGPU metadata blob
+  char sampleMetadata[494] = {
+      -126, -80, 97,  109, 100,  112, 97,   108,  46,  112,  105,  112,  101,  108, 105,  110, 101, 115,  -111, -119,
+      -92,  46,  97,  112, 105,  -90, 86,   117,  108, 107,  97,   110,  -80,  46,  104,  97,  114, 100,  119,  97,
+      114,  101, 95,  115, 116,  97,  103,  101,  115, -127, -93,  46,   99,   115, -122, -84, 46,  101,  110,  116,
+      114,  121, 95,  112, 111,  105, 110,  116,  -81, 95,   97,   109,  100,  103, 112,  117, 95,  99,   115,  95,
```

Comment:
This seems unmaintainable. Could you use the Document writer to parse a yaml and then serialize it to a StringRef?

Reply:
Ah, that's much better, I didn't realize that output was already in a markup format. Thanks!